src/plugins/intel_gna/src/pre_post_process/transposition_info.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+#include "openvino/core/model.hpp"
+
+namespace ov {
+namespace intel_gna {
+namespace pre_post_process {
+
+struct TranspositionInfo {
+    bool transpose;
+    size_t num_transpose_rows;
+    size_t num_transpose_columns;
+};
+
+using TranspositionInfoMap = std::map<std::string, std::vector<TranspositionInfo>>;
+
+/*
+ * Convert transposition info to preprocessing model
+ */
```

Comment:
Please extend the description here and it won't be necessary in the cpp file then.

Reply:
done